pkg/node/islb/internal.go

```
@@ -393,41 +397,52 @@ func broadcast(data map[string]interface{}) (map[string]interface{}, *nprotoo.Er
 	return util.Map(), nil
 }
 
+func handle(msg map[string]interface{}) (map[string]interface{}, *nprotoo.Error) {
+	method := msg["method"].(string)
+	data := msg["data"].(map[string]interface{})
+	log.Infof("method => %s, data => %v", method, data)
+
+	switch method {
+	case proto.IslbFindService:
+		return findServiceNode(data)
+	case proto.IslbOnStreamAdd:
+		return streamAdd(data)
+	case proto.IslbOnStreamRemove:
+		return streamRemove(data)
+	case proto.IslbGetPubs:
+		return getPubs(data)
+	case proto.IslbClientOnJoin:
+		return clientJoin(data)
+	case proto.IslbClientOnLeave:
+		return clientLeave(data)
+	case proto.IslbGetMediaInfo:
+		return getMediaInfo(data)
+	case proto.IslbRelay:
+		return relay(data)
+	case proto.IslbUnrelay:
+		return unRelay(data)
+	case proto.IslbOnBroadcast:
+		return broadcast(data)
+	}
+
+	return nil, util.NewNpError(400, fmt.Sprintf("Unkown method [%s]", method))
+}
+
+func handleBroadcast(msg map[string]interface{}, subj string) {
+	go func(msg map[string]interface{}) {
+		_, err := handle(msg)
+		if err != nil {
+			log.Errorf("handleBroadcast error => %s", err)
```

Maybe this should silently fail (since it isn't expected that islb can handle all events).